### PR TITLE
Remove default params in query that are empty lists - temporarily remove octopart params

### PR DESCRIPTION
**Pull request by PhilippeFerreiraDeSousa**
I think I need to see what is going on with the octopart logic in the backend when stock != 0, in the mean time we can use stock=0 by default which won't call the octopart logic by default (thanks to jitx-client PR https://github.com/JITx-Inc/jitx-client/pull/356 that removes default sellers).

This will also remove the stock requirements on resistors / inductors / capacitors which was useful.
Will revert default stock when I find out if something is off with the octopart logic, will probably reduce it as 100 is small for resistors but a lot for mcus.

### CODE REVIEW

File: utils/micro-controllers.stanza [hunk 1]
```diff
@@ -97,33 +97,34 @@ public defn MicroController (properties:Tuple<KeyValue>) -> MicroController :
...
-                                            "_stock" => DESIGN-QUANTITY],
-                                          vendor-properties
-                                          temperature-properties,
-                                          properties])
+
+  val octopart-parameters = Vector<KeyValue<String, ?>>()
+  if DESIGN-QUANTITY > 0 :
+    add(octopart-parameters, "_stock" => DESIGN-QUANTITY)
+  match(vendors: Tuple<String>) :
+    if length(vendors) != 0 :
```
Comment (bpellenard) [inline utils/micro-controllers.stanza:110]: `empty?`?

**Review by bpellenard — APPROVED** [review]
Looks good to me!

File: utils/micro-controllers.stanza [hunk 2]
```diff
@@ -97,33 +97,34 @@ public defn MicroController (properties:Tuple<KeyValue>) -> MicroController :
...
-  else :
-    query-properties = to-tuple $ cat-all([["category" => "microcontroller",
-                                            "_sort" => sort,
-                                            "_exist" => exist,
-                                            "_stock" => DESIGN-QUANTITY],
-                                          vendor-properties
-                                          temperature-properties,
-                                          properties])
+
+  val octopart-parameters = Vector<KeyValue<String, ?>>()
```
Comment (m-hilgendorf) [inline utils/micro-controllers.stanza:106]: perfect place for a (stanza) generator. 
```
val optional-properties =
  generate<KeyValue> : 
    yield("_stock" => DESIGN-QUANTITY) when DESIGN-QUANTITY > 0
    yield("_sellers" => vendors) when call?<Tuple>(empty?, vendors) is-not False ; import lang-utils
    yield("_sort" => sort) when not empty?(sort)
    yield("_exist" => exist) when not empty?(exist)
    match(operating-temperature:[Double, Double]):
      yield("max-rated-temperature.min" => operating-temperature[0])
      yield("min-rated-temperature.max" => operating-temperature[1])

  val query-properties = to-tuple(cat-all([properties, optional-properties]))
  val result = dbquery-all(query-properties) as JObject
  throw(NoComponentMeetingRequirements(query-properties)) when empty?(result)
  result 
```

Comment (PhilippeFerreiraDeSousa) [reply utils/micro-controllers.stanza:103]: Cool trick the `call?`

File: designs/mcu.stanza [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #use-added-syntax(jitx)
-defpackage ocdb/designs/tutorial :
+defpackage ocdb/designs/mcu :
```
Comment (ariel-hi) [inline designs/mcu.stanza:2]: 👍

**Review by ariel-hi — APPROVED** [review]
Awesome

**Pull request merged**